internal/slicer/slicer_test.go

```diff
@@ -1051,6 +1051,42 @@ var slicerTests = []slicerTest{{
 						content.list("/foo-bar/")
 		`,
 	},
+}, {
+	summary: "Valid same hard link in two slices in the same package",
+	slices: []setup.SliceKey{
+		{"test-package", "slice1"},
+		{"test-package", "slice2"}},
+	pkgs: map[string][]byte{
+		"test-package": testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Dir(0755, "./dir/"),
+			testutil.Reg(0644, "./dir/file", "text for file"),
+			testutil.Hln(0644, "./hardlink", "./dir/file"),
+		}),
+	},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				slice1:
+					contents:
+						/dir/file:
+						/hardlink:
+				slice2:
+					contents:
+						/dir/file:
+						/hardlink:
+		`,
+	},
+	filesystem: map[string]string{
+		"/dir/":     "dir 0755",
+		"/dir/file": "file 0644 28121945",
+		"/hardlink": "file 0644 28121945",
+	},
+	report: map[string]string{
+		"/dir/file": "file 0644 28121945 {test-package_slice1,test-package_slice2}",
+		"/hardlink": "hardlink 0644 /dir/file {test-package_slice1,test-package_slice2}",
```

Comment:
One last thing Jame, I think the mode here is not adding anything because it has to be the exact same one as the file it references.

Reply:
Updated in [4313070](https://github.com/letFunny/chisel/pull/17/commits/4313070b5a0af67b4c1cdca323c0eb03d7f882b9).